feature/BTSLA-26-backend-evolution

sergiopj opened the pull request.

Commit:
removal of the description field in the Game entity, with all that it entails in the app

Commit:
added modification in the home and game screens, removed part of the session data in the home dialog

Commit:
the following functionality was modified:
- request for data to firebase by get in json format, with in config and client
- modification in the models
- modification in the dialogs to adapt the data to return to the webapp

dcantelar commented on webapps/set-top-box-haac/src/screens/game/index.tsx:
```diff
@@ -1,36 +1,35 @@
 import { useAura } from '@telefonica/la-web-sdk';
 
-const GameScreen: React.FC<GameScreenData> = (gameData: GameScreenData) => {
+const GameScreen: React.FC<GameScreenData> = (data: any) => {
```
este `any` hay q evitarlo aqui, este ejemplo deberia tender a convertirse en nuestra livingapp _insignia_, con lo q deberia incluir las mejores practicas... aqui debe llegar el modelo d esta pantalla, algo pasa por aqui...

dcantelar commented on webapps/set-top-box-haac/src/screens/home/index.tsx:
```diff
@@ -6,8 +6,8 @@ import { useAura } from '@telefonica/la-web-sdk';
 
-const HomeScreen: React.FC<HomeScreenData> = (data: HomeScreenData) => {
-    const { categories, title } = data;
+const HomeScreen: React.FC<HomeScreenData> = (data: any) => {
```
hum, este any...

Commit:
adapted the backend to the new video game store architecture with several games carousels grouped by platform and category
both the front and much of the backend is still unfinished

Commit:
added the following functionality:
- data model and modified data, gameCards entity is added
- the dialog is added with all its logic
- adapted models and customer for new functionality

Commit:
evolution of the shopping cart dialog

Commit:
modified cartGame models

Commit:
fix
